ZukoAzula2016/src/org/team1540/zukoazula/ShooterSpeedControl.java

```diff
@@ -0,0 +1,55 @@
+package org.team1540.zukoazula;
+
+import ccre.channel.BooleanInput;
+import ccre.channel.EventCell;
+import ccre.channel.EventInput;
+import ccre.channel.FloatCell;
+import ccre.channel.FloatInput;
+import ccre.channel.FloatOutput;
+import ccre.cluck.Cluck;
+import ccre.ctrl.ExtendedMotor;
+import ccre.ctrl.ExtendedMotorFailureException;
+import ccre.frc.FRC;
+import ccre.time.Time;
+
+public class ShooterSpeedControl {
+    private static final ExtendedMotor shooterCAN = FRC.talonCAN(7);
+    private static BooleanInput shooting;
+    private static FloatOutput shooterMotor;
+    private static FloatInput gearTooth;
+
+    private static final FloatCell velocity = new FloatCell();
+
+    public static void setup() throws ExtendedMotorFailureException {
+        shooting = ZukoAzula.controlBinding.addBoolean("Shoot");
+        shooterMotor = shooterCAN.simpleControl();
+        EventInput shootingPressed = shooting.onPress();
+        gearTooth = FRC.encoder(0, 0, false, shootingPressed);
+        velocityOf(gearTooth, shootingPressed).send(velocity);
+        shooting.toFloat(0, (bangBangControl(velocity, ZukoAzula.mainTuning.getFloat("Shooter Velocity Target", 1), 0, 1))).send(shooterMotor);
```

Comment:
Consider integrating enabling/disabling of the shooter and the velocity calculation into the bang bang controller. (Not necessarily inlining anything, but try for more encapsulation.)